Source/SwiftLintFramework/Rules/Lint/DuplicatedKeyInDictionaryLiteralRule.swift

```diff
@@ -79,74 +79,59 @@ public struct DuplicatedKeyInDictionaryLiteralRule: ASTRule, ConfigurationProvid
         ]
     )
 
-    public func validate(file: SwiftLintFile, kind: SwiftExpressionKind,
-                         dictionary: SourceKittenDictionary) -> [StyleViolation] {
-        guard kind == .dictionary else {
-            return []
-        }
-
-        let keys = nonGeneratedDictionaryKeys(with: file, dictionary: dictionary)
-        guard keys.count >= 2 else {
-            return []
-        }
+    public func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
+        Visitor(viewMode: .sourceAccurate)
+    }
+}
 
-        var existingKeys: [String: DictionaryKey] = [:]
-        return keys
-            .filter { key in
-                guard let existingKey = existingKeys[key.content] else {
-                    existingKeys[key.content] = key
-                    return false
+private extension DuplicatedKeyInDictionaryLiteralRule {
+    final class Visitor: ViolationsSyntaxVisitor {
+        override func visitPost(_ list: DictionaryElementListSyntax) {
+            let keys = list.map(\.keyExpression).compactMap { expr -> DictionaryKey? in
+                expr.stringContent.map {
+                    DictionaryKey(position: expr.positionAfterSkippingLeadingTrivia, content: $0)
                 }
-
-                let existingKeyKinds = file.syntaxMap.kinds(inByteRange: existingKey.byteRange)
-                let keyKinds = file.syntaxMap.kinds(inByteRange: key.byteRange)
```

Comment:
we're not doing this check anymore, which in theory could lead to false positives, but we also don't do it in other rules like `identical_operands` so we should be fine?